packages/core/strapi/lib/services/entity-service/types/params/publication-state.d.ts

```diff
@@ -0,0 +1,30 @@
+import type { Common, Utils } from '@strapi/strapi';
+
+export type Kind = 'preview' | 'live';
+
+export type IsEnabled<TSchemaUID extends Common.UID.Schema> = Utils.Expression.MatchFirst<
+  [
+    [
+      Common.UID.IsContentType<TSchemaUID>,
+      Utils.Expression.IsTrue<Common.Schemas[TSchemaUID]['options']['draftAndPublish']>
```

Comment:
When the types are not generated this will be failing here instead of falling back to Truthy

Reply:
Adding more context here:
- When passing Common.UID.Schema, this will work as intended because it won't pass the test for both ContentType || Component (as it represents both at the same time)
- However, when passing a raw UID string, it'll match it to a content-type first and try to find a value for the `options.draftAndPublish` in a loose ContentType schema (from the shared registry default signature). Since it won't find a truthy value, it'll return that draftAndPublish is not enabled.

Reply:
![image](https://github.com/strapi/strapi/assets/25851739/dab4931b-09e0-4846-b31f-9f98cb1e38f5)

Reply:
tbh I'm not sure about how A8 works but not A9 or A10, since they're all part of the UID.ContentType union

Reply:
![image](https://github.com/strapi/strapi/assets/25851739/c38ddc5f-ffd6-438c-ad0d-9f4bee4243ce)
![image](https://github.com/strapi/strapi/assets/25851739/49942e47-a7fe-42d2-a1d5-9e1018de8d50)

Reply:
@alexandrebodin fixed in [795f761](https://github.com/strapi/strapi/pull/17281/commits/795f761c01441d6990d5318a43fd092bb0434dbb)